web/src/compose_ui.js

```diff
@@ -495,6 +527,50 @@ export function format_text($textarea, type, inserted_content) {
 
             wrapSelection(field, italic_syntax);
             break;
+        case "bulleted": {
+            // Following behaviour is inspired from GitHub
+            let {before_lines, selected_lines, after_lines} = section_off_selected_lines();
+            // if there is even a single unbulleted line selected, we bullet all
+            const should_bullet = selected_lines
+                .split("\n")
+                .some((line) => !list_util.is_bulleted(line));
+            if (should_bullet) {
+                selected_lines = selected_lines
+                    .split("\n")
+                    .map((line) => "- " + line)
+                    .join("\n");
+                // add blank line between text before and list if not already present
+                if (before_lines.length && before_lines.at(-2) !== "\n") {
+                    before_lines += "\n";
+                }
+                // add blank line between list and rest of text if not already present
+                if (after_lines.length && after_lines.at(1) !== "\n") {
```

Comment:
Why is this `.at(1)` rather than `.at(0)`?

Reply:
This is due to similar logic as the previous comment (`after_lines` always has the separating new line at its start)